go-controller/pkg/ovn/ovn.go

```diff
@@ -609,6 +638,125 @@ func (oc *Controller) GetServiceVIPToName(vip string, protocol kapi.Protocol) (t
 	return namespace, ok
 }
 
+// AddServiceLBToACL associates an empty load balancer with its associated ACL reject rule
+func (oc *Controller) SetServiceACLToLB(lb, vip, acl string) {
+	oc.serviceLBLock.Lock()
+	defer oc.serviceLBLock.Unlock()
+	if _, ok := oc.serviceLBMap[lb]; !ok {
+		oc.serviceLBMap[lb] = make(map[string]*LoadBalancerConf)
+		oc.serviceLBMap[lb][vip] = &LoadBalancerConf{RejectACL: acl}
+		return
+	}
+	if _, ok := oc.serviceLBMap[lb][vip]; !ok {
+		oc.serviceLBMap[lb][vip] = &LoadBalancerConf{RejectACL: acl}
+		return
+	}
+	// This should not happen, the entry should be removed for the VIP when the service is deleted for an existing
+	// entry
+	if oc.serviceLBMap[lb][vip] == nil {
+		klog.Errorf("config for existing load balancer is nil. LB: %s, VIP: %s", lb, vip)
+		oc.serviceLBMap[lb][vip] = &LoadBalancerConf{RejectACL: acl}
+		return
+	}
+	oc.serviceLBMap[lb][vip].RejectACL = acl
+}
+
+// SetServiceEndpointsToLB associates a load balancer with endpoints
+func (oc *Controller) SetServiceEndpointsToLB(lb, vip string, eps []string) {
+	oc.serviceLBLock.Lock()
+	defer oc.serviceLBLock.Unlock()
+	if _, ok := oc.serviceLBMap[lb]; !ok {
+		oc.serviceLBMap[lb] = make(map[string]*LoadBalancerConf)
+		oc.serviceLBMap[lb][vip] = &LoadBalancerConf{Endpoints: eps}
+		return
+	}
+	if _, ok := oc.serviceLBMap[lb][vip]; !ok {
+		oc.serviceLBMap[lb][vip] = &LoadBalancerConf{Endpoints: eps}
+		return
+	}
+	// This should not happen, the entry should be removed for the VIP when the service is deleted
```

Comment:
@trozet I don't think this ever can happen, unless we're explicitly setting the map entry to `nil` which the code doesn't do anywhere that I can see. Let's kill the block.

Reply:
I agree it shouldn't happen (that's why I added the comment), but this protects us from a from a potential panic down the road. imho 3 lines of extra protection against a crash, no matter how improbable, is worth it, but if you strongly disagree then I will remove it.

Reply:
@trozet yeah, I think we should just remove it. While only ~15 LoC we usually don't try to aggressively protect ourselves from things that our code shouldn't be doing anyway.

Reply:
ok I will remove it in both places